java/undertow/src/main/java/the/benchmarker/Main.java

```diff
@@ -0,0 +1,44 @@
+package the.benchmarker;
+
+import io.undertow.Undertow;
+import io.undertow.server.HttpHandler;
+import io.undertow.server.HttpServerExchange;
+import io.undertow.server.RoutingHandler;
+import io.undertow.util.Headers;
+
+import java.util.Deque;
+import java.util.Optional;
+
+public class Main {
+
+    public static void main(final String[] args) {
+        Undertow server = Undertow.builder()
+                .addHttpListener(3000, "0.0.0.0")
+                .setHandler(new RoutingHandler()
+                        .get("/", new HttpHandler() {
+                            @Override
+                            public void handleRequest(HttpServerExchange exchange) throws Exception {
+                                exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, "text/plain");
+                                exchange.getResponseSender().send("");
+                            }
+                        })
+                        .post("/user", new HttpHandler() {
+                            @Override
+                            public void handleRequest(HttpServerExchange exchange) throws Exception {
+                                exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, "text/plain");
+                                exchange.getResponseSender().send("");
+                            }
+                        })
+                        .get("/user/{name}", new HttpHandler() {
+                            @Override
+                            public void handleRequest(HttpServerExchange exchange) throws Exception {
+                                exchange.getResponseHeaders().put(Headers.CONTENT_TYPE, "text/plain");
+                                var x = Optional.ofNullable(exchange.getQueryParameters().get("name"))
```

Comment:
Note that using lambdas and functional programming here will be slower than just manually checking for null.

Reply:
I'm curious to know more about how this "littlle" thing can slow down this endpoint (curious in general, not for this specific case). Care to elaborate further?

Reply:
Basically just extra allocations when you don't really need them. Undertow has been optimised to avoid allocation as much as possible, so for these super simple style tests extra allocation inside the handler may make a noticeable difference. 

It's less of an issue in a 'real' app because the app generally generates a lot more garbage doing business logic, so a few extra allocations are unnoticeable.